server/src/main/java/org/elasticsearch/search/vectors/KnnVectorQueryBuilder.java

```diff
@@ -251,7 +350,8 @@ protected QueryBuilder doRewrite(QueryRewriteContext ctx) throws IOException {
             rewrittenQueries.add(rewrittenQuery);
         }
         if (changed) {
-            return new KnnVectorQueryBuilder(fieldName, queryVector, numCands, vectorSimilarity).boost(boost)
+            return new KnnVectorQueryBuilder(fieldName, queryVector, queryVectorBuilder, queryVectorSupplier, numCands, vectorSimilarity)
```

Comment:
`queryVectorBuilder` and `queryVectorSupplier ` are always null here right? Maybe we don't need this ctor?